app/reports/views.py

```diff
@@ -0,0 +1,32 @@
+from flask import render_template
+from flask.ext.login import login_required, current_user
+from . import reports
+from ..models import IncidentReport, Agency
+
+
+@reports.route('/all')
+@login_required
+def view_reports():
+    """View all idling incident reports.
+    Different roles have access to different reports.
+    An Admin can see all reports.
+    An Agency worker can all reports for their agency.
+    A general user can see all reports submitted by that user."""
+
+    if current_user.is_admin():
+        reports = IncidentReport.query.all()
+        agencies = Agency.query.all()
+
+    elif current_user.is_agency_worker():
+        reports = []
+        for agency in current_user.agencies:
+            reports.extend(agency.incident_reports)
+        agencies = None
```

Comment:
I would set agencies to None before the if/elif/else block. This takes advantage of what I consider to be a confusing and peculiar feature of python.